src/text-editor.coffee

```diff
@@ -2587,6 +2587,16 @@ class TextEditor extends Model
       maintainClipboard = true
     return
 
+
+  # Essential: For each selection, only copy hightlighted text. Line is coptied if no selected text and cursor is at the beginning of the line
+  copyOnlySelectedText: ->
+    maintainClipboard = false
+    for selection in @getSelectionsOrderedByBufferPosition()
+      if not selection.isEmpty()
+        selection.copy(maintainClipboard, true)
+      maintainClipboard = true
```

Comment:
I would think that the `maintainClipboard = true` statement should be *inside* of the `if not selection.isEmpty()` block. Otherwise, if the first selection is empty and the second selection is not, the existing clipboard entry will be mutated unexpectedly.